Things are not progressing as I expected, so here's the stacktrace I am seeing with the current version of this PR. I tried a lot of variations (including recomputing `Inet.INET6_ANY`, `Inet.INET6_LOOPBACK`, `CidrAddress.INET4_ANY_CIDR` and `CidrAddress.INET6_ANY_CIDR`)  without success so far. I'm still trying to make this work, but I'm open to any suggestion :)

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on OpenJDK 64-Bit GraalVM 20.0.0-dev
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] /Library/Java/JavaVirtualMachines/graalvm-ce-19.3.0/Contents/Home/jre/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -H:+TraceClassInitialization --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar config-quickstart-1.0-SNAPSHOT-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http -H:+JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace config-quickstart-1.0-SNAPSHOT-runner
[config-quickstart-1.0-SNAPSHOT-runner:47596]    classlist:  21,220.81 ms
[config-quickstart-1.0-SNAPSHOT-runner:47596]        (cap):   8,132.22 ms
[config-quickstart-1.0-SNAPSHOT-runner:47596]        setup:  13,471.06 ms
22:31:06,770 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Final
[config-quickstart-1.0-SNAPSHOT-runner:47596]   (typeflow):  44,098.86 ms
[config-quickstart-1.0-SNAPSHOT-runner:47596]    (objects):  25,314.82 ms
[config-quickstart-1.0-SNAPSHOT-runner:47596]   (features):   1,551.88 ms
[config-quickstart-1.0-SNAPSHOT-runner:47596]     analysis:  73,797.01 ms
Error: Unsupported features in 2 methods
Detailed message:
Error: No instances of java.net.Inet4Address are allowed in the image heap as this class should be initialized at image runtime. Object has been initialized without the native-image initialization instrumentation and the stack trace can't be tracked.
Trace: Object was reached by 
	reading field org.wildfly.common.net.Inet.INET4_BROADCAST
Error: No instances of java.net.Inet6Address are allowed in the image heap as this class should be initialized at image runtime. Object has been initialized without the native-image initialization instrumentation and the stack trace can't be tracked.
Trace: Object was reached by 
	reading field org.wildfly.common.net.CidrAddress.networkAddress of
		constant org.wildfly.common.net.CidrAddress@50a9de01 reached by 
	scanning method org.wildfly.common.net.CidrAddress.create(CidrAddress.java:127)
Call path from entry point to org.wildfly.common.net.CidrAddress.create(byte[], int, boolean): 
	at org.wildfly.common.net.CidrAddress.create(CidrAddress.java:116)
	at com.oracle.svm.reflect.CidrAddress_create_303e462200a7304a7ae6e75c332c52433511f071_166.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.enterprise.util.AnnotationLiteral.invoke(AnnotationLiteral.java:288)
	at javax.enterprise.util.AnnotationLiteral.getMemberValue(AnnotationLiteral.java:276)
	at javax.enterprise.util.AnnotationLiteral.hashCode(AnnotationLiteral.java:246)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at com.oracle.svm.jni.access.JNIReflectionDictionary.getFieldNameByID(JNIReflectionDictionary.java:278)
	at com.oracle.svm.jni.functions.JNIFunctions.ToReflectedField(JNIFunctions.java:844)
	at com.oracle.svm.core.code.IsolateEnterStub.JNIFunctions_ToReflectedField_80d8233579d5215df0227b770e5c01228a0de9b9(generated:0)

com.oracle.svm.core.util.UserError$UserException: Unsupported features in 2 methods
Detailed message:
Error: No instances of java.net.Inet4Address are allowed in the image heap as this class should be initialized at image runtime. Object has been initialized without the native-image initialization instrumentation and the stack trace can't be tracked.
Trace: Object was reached by 
	reading field org.wildfly.common.net.Inet.INET4_BROADCAST
Error: No instances of java.net.Inet6Address are allowed in the image heap as this class should be initialized at image runtime. Object has been initialized without the native-image initialization instrumentation and the stack trace can't be tracked.
Trace: Object was reached by 
	reading field org.wildfly.common.net.CidrAddress.networkAddress of
		constant org.wildfly.common.net.CidrAddress@50a9de01 reached by 
	scanning method org.wildfly.common.net.CidrAddress.create(CidrAddress.java:127)
Call path from entry point to org.wildfly.common.net.CidrAddress.create(byte[], int, boolean): 
	at org.wildfly.common.net.CidrAddress.create(CidrAddress.java:116)
	at com.oracle.svm.reflect.CidrAddress_create_303e462200a7304a7ae6e75c332c52433511f071_166.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.enterprise.util.AnnotationLiteral.invoke(AnnotationLiteral.java:288)
	at javax.enterprise.util.AnnotationLiteral.getMemberValue(AnnotationLiteral.java:276)
	at javax.enterprise.util.AnnotationLiteral.hashCode(AnnotationLiteral.java:246)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at com.oracle.svm.jni.access.JNIReflectionDictionary.getFieldNameByID(JNIReflectionDictionary.java:278)
	at com.oracle.svm.jni.functions.JNIFunctions.ToReflectedField(JNIFunctions.java:844)
	at com.oracle.svm.core.code.IsolateEnterStub.JNIFunctions_ToReflectedField_80d8233579d5215df0227b770e5c01228a0de9b9(generated:0)

	at com.oracle.svm.core.util.UserError.abort(UserError.java:75)
	at com.oracle.svm.hosted.FallbackFeature.reportAsFallback(FallbackFeature.java:221)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:735)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:529)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:444)
	at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Unsupported features in 2 methods
Detailed message:
Error: No instances of java.net.Inet4Address are allowed in the image heap as this class should be initialized at image runtime. Object has been initialized without the native-image initialization instrumentation and the stack trace can't be tracked.
Trace: Object was reached by 
	reading field org.wildfly.common.net.Inet.INET4_BROADCAST
Error: No instances of java.net.Inet6Address are allowed in the image heap as this class should be initialized at image runtime. Object has been initialized without the native-image initialization instrumentation and the stack trace can't be tracked.
Trace: Object was reached by 
	reading field org.wildfly.common.net.CidrAddress.networkAddress of
		constant org.wildfly.common.net.CidrAddress@50a9de01 reached by 
	scanning method org.wildfly.common.net.CidrAddress.create(CidrAddress.java:127)
Call path from entry point to org.wildfly.common.net.CidrAddress.create(byte[], int, boolean): 
	at org.wildfly.common.net.CidrAddress.create(CidrAddress.java:116)
	at com.oracle.svm.reflect.CidrAddress_create_303e462200a7304a7ae6e75c332c52433511f071_166.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at javax.enterprise.util.AnnotationLiteral.invoke(AnnotationLiteral.java:288)
	at javax.enterprise.util.AnnotationLiteral.getMemberValue(AnnotationLiteral.java:276)
	at javax.enterprise.util.AnnotationLiteral.hashCode(AnnotationLiteral.java:246)
	at java.util.HashMap.hash(HashMap.java:339)
	at java.util.HashMap.get(HashMap.java:557)
	at com.oracle.svm.jni.access.JNIReflectionDictionary.getFieldNameByID(JNIReflectionDictionary.java:278)
	at com.oracle.svm.jni.functions.JNIFunctions.ToReflectedField(JNIFunctions.java:844)
	at com.oracle.svm.core.code.IsolateEnterStub.JNIFunctions_ToReflectedField_80d8233579d5215df0227b770e5c01228a0de9b9(generated:0)

	at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:129)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:732)
	... 7 more
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigurationSetup.java
@@ -77,6 +77,7 @@ import io.quarkus.runtime.configuration.HyphenateEnumConverter;
 import io.quarkus.runtime.configuration.NameIterator;
 import io.quarkus.runtime.configuration.ProfileManager;
 import io.quarkus.runtime.configuration.SimpleConfigurationProviderResolver;
+import io.quarkus.runtime.graal.InetRunTime;
 import io.smallrye.config.Converters;
 import io.smallrye.config.SmallRyeConfig;
 import io.smallrye.config.SmallRyeConfigBuilder;
@@ -820,5 +821,10 @@ public class ConfigurationSetup {
 
     ) {
 
+    }
+
+    @BuildStep
+    RuntimeInitializedClassBuildItem runtimeInitializedClass() {
+        return new RuntimeInitializedClassBuildItem(InetRunTime.class.getName());
     }
 }

--- a/core/runtime/src/main/java/io/quarkus/runtime/graal/CidrAddressSubstitutions.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/graal/CidrAddressSubstitutions.java
@@ -1,0 +1,91 @@
+package io.quarkus.runtime.graal;
+
+import java.net.InetAddress;
+
+import org.wildfly.common.net.CidrAddress;
+import org.wildfly.common.net.Inet;
+
+import com.oracle.svm.core.SubstrateUtil;
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.InjectAccessors;
+import com.oracle.svm.core.annotate.TargetClass;
+
+import io.quarkus.runtime.graal.Target_org_wildfly_common_net_CidrAddress.CidrAddressUtil;
+
+/*
+ * The following substitutions are required because of a new restriction in GraalVM 19.3.0 that prohibits the presence of
+ * java.net.Inet4Address and java.net.Inet6Address in the image heap. Each field annotated with @InjectAccessors is lazily
+ * recomputed at runtime on first access while CidrAddress.class can still be initialized during the native image build.
+ */
+@TargetClass(CidrAddress.class)
+final class Target_org_wildfly_common_net_CidrAddress {
+
+    @Alias
+    private Target_org_wildfly_common_net_CidrAddress(InetAddress networkAddress, int netmaskBits) {
+    }
+
+    @Alias
+    @InjectAccessors(Inet4AnyCidrAccessor.class)
+    public static CidrAddress INET4_ANY_CIDR;
+
+    @Alias
+    @InjectAccessors(Inet6AnyCidrAccessor.class)
+    public static CidrAddress INET6_ANY_CIDR;
+
+    static class CidrAddressUtil {
+        static CidrAddress newInstance(InetAddress networkAddress, int netmaskBits) {
+            return SubstrateUtil.cast(new Target_org_wildfly_common_net_CidrAddress(networkAddress, netmaskBits),
+                    CidrAddress.class);
+        }
+    }
+}
+
+class Inet4AnyCidrAccessor {
+
+    private static volatile CidrAddress INET4_ANY_CIDR;
+
+    static CidrAddress get() {
+        CidrAddress result = INET4_ANY_CIDR;
+        if (result == null) {
+            // Lazy initialization on first access.
+            result = initializeOnce();
+        }
+        return result;
+    }
+
+    private static synchronized CidrAddress initializeOnce() {
+        CidrAddress result = INET4_ANY_CIDR;
+        if (result != null) {
+            // Double-checked locking is OK because INSTANCE is volatile.
+            return result;
+        }
+        result = CidrAddressUtil.newInstance(Inet.INET4_ANY, 0);
+        INET4_ANY_CIDR = result;
+        return result;
+    }
+}
+
+class Inet6AnyCidrAccessor {
+
+    private static volatile CidrAddress INET6_ANY_CIDR;
+
+    static CidrAddress get() {
+        CidrAddress result = INET6_ANY_CIDR;
+        if (result == null) {
+            // Lazy initialization on first access.
+            result = initializeOnce();
+        }
+        return result;
+    }
+
+    private static synchronized CidrAddress initializeOnce() {
+        CidrAddress result = INET6_ANY_CIDR;
+        if (result != null) {
+            // Double-checked locking is OK because INSTANCE is volatile.
+            return result;
+        }
+        result = CidrAddressUtil.newInstance(Inet.INET6_ANY, 0);
+        INET6_ANY_CIDR = result;
+        return result;
+    }
+}

--- a/core/runtime/src/main/java/io/quarkus/runtime/graal/InetRunTime.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/graal/InetRunTime.java
@@ -1,0 +1,14 @@
+package io.quarkus.runtime.graal;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+
+import org.wildfly.common.net.Inet;
+
+public class InetRunTime {
+    public static final Inet4Address INET4_ANY = Inet.getInet4Address(0, 0, 0, 0);
+    public static final Inet4Address INET4_LOOPBACK = Inet.getInet4Address(127, 0, 0, 1);
+    public static final Inet4Address INET4_BROADCAST = Inet.getInet4Address(255, 255, 255, 255);
+    public static final Inet6Address INET6_ANY = Inet.getInet6Address(0, 0, 0, 0, 0, 0, 0, 0);
+    public static final Inet6Address INET6_LOOPBACK = Inet.getInet6Address(0, 0, 0, 0, 0, 0, 0, 1);
+}

--- a/core/runtime/src/main/java/io/quarkus/runtime/graal/InetSubstitutions.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/graal/InetSubstitutions.java
@@ -1,0 +1,69 @@
+package io.quarkus.runtime.graal;
+
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+
+import org.wildfly.common.net.Inet;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.InjectAccessors;
+import com.oracle.svm.core.annotate.TargetClass;
+
+/*
+ * The following substitutions are required because of a new restriction in GraalVM 19.3.0 that prohibits the presence of
+ * java.net.Inet4Address and java.net.Inet6Address in the image heap. Each field annotated with @InjectAccessors is lazily
+ * recomputed at runtime on first access while Inet.class can still be initialized during the native image build.
+ */
+@TargetClass(Inet.class)
+final class Target_org_wildfly_common_net_Inet {
+
+    @Alias
+    @InjectAccessors(Inet4AnyAccessor.class)
+    public static Inet4Address INET4_ANY;
+
+    @Alias
+    @InjectAccessors(Inet4LoopbackAccessor.class)
+    public static Inet4Address INET4_LOOPBACK;
+
+    @Alias
+    @InjectAccessors(Inet4BroadcastAccessor.class)
+    public static Inet4Address INET4_BROADCAST;
+
+    @Alias
+    @InjectAccessors(Inet6AnyAccessor.class)
+    public static Inet6Address INET6_ANY;
+
+    @Alias
+    @InjectAccessors(Inet6LoopbackAccessor.class)
+    public static Inet6Address INET6_LOOPBACK;
+}
+
+final class Inet4AnyAccessor {
+    static Inet4Address get() {
+        return InetRunTime.INET4_ANY;
+    }
+}
+
+final class Inet4LoopbackAccessor {
+    static Inet4Address get() {
+        return InetRunTime.INET4_LOOPBACK;
+    }
+}
+
+final class Inet4BroadcastAccessor {
+    static Inet4Address get() {
+        return InetRunTime.INET4_BROADCAST;
+    }
+}
+
+final class Inet6AnyAccessor {
+    static Inet6Address get() {
+        return InetRunTime.INET6_ANY;
+    }
+}
+
+final class Inet6LoopbackAccessor {
+    static Inet6Address get() {
+        return InetRunTime.INET6_LOOPBACK;
+    }
+}

--- a/integration-tests/main/src/main/java/io/quarkus/it/config/MicroProfileConfigResource.java
+++ b/integration-tests/main/src/main/java/io/quarkus/it/config/MicroProfileConfigResource.java
@@ -6,6 +6,7 @@ import javax.ws.rs.Path;
 
 import org.eclipse.microprofile.config.Config;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.wildfly.common.net.CidrAddress;
 
 /**
  * Test some MicroProfile config primitives.
@@ -18,6 +19,9 @@ public class MicroProfileConfigResource {
 
     @ConfigProperty(name = "microprofile.custom.value")
     MicroProfileCustomValue value;
+
+    @ConfigProperty(name = "microprofile.cidr-address")
+    CidrAddress cidrAddress;
 
     @Inject
     Config config;
@@ -35,5 +39,11 @@ public class MicroProfileConfigResource {
     @Path("/get-custom-value")
     public String getCustomValue() {
         return Integer.toString(value.getNumber());
+    }
+
+    @GET
+    @Path("/get-cidr-address")
+    public String getCidrAddress() {
+        return cidrAddress.getNetworkAddress().getHostAddress();
     }
 }

--- a/integration-tests/main/src/main/resources/application.properties
+++ b/integration-tests/main/src/main/resources/application.properties
@@ -26,6 +26,7 @@ names=quarkus,redhat
 schedulerservice.cron.expr=0/10 * * * * ?
 
 microprofile.custom.value = 456
+microprofile.cidr-address=10.0.0.20/24
 configproperties.message=Hello
 configproperties.other=1
 

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/MicroProfileConfigTestCase.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/MicroProfileConfigTestCase.java
@@ -25,4 +25,11 @@ public class MicroProfileConfigTestCase {
                 .when().get("/microprofile-config/get-custom-value")
                 .then().body(is("456"));
     }
+
+    @Test
+    public void testCidrAddress() {
+        RestAssured
+                .when().get("/microprofile-config/get-cidr-address")
+                .then().body(is("10.0.0.0"));
+    }
 }

